framework/packages/abstract-client/src/account.rs

```diff
@@ -0,0 +1,162 @@
+use abstract_core::{
+    manager::{state::AccountInfo, InfoResponse},
+    objects::{gov_type::GovernanceDetails, namespace::Namespace, AssetEntry},
+    version_control::NamespaceResponse,
+};
+use abstract_interface::{
+    Abstract, AbstractAccount, AccountDetails, ManagerQueryFns, RegisteredModule, VCQueryFns,
+};
+use cw_orch::contract::Contract;
+use cw_orch::prelude::*;
+use serde::Serialize;
+
+use crate::{
+    application::Application, client::AbstractClientResult, infrastructure::Infrastructure,
+};
+
+pub struct AccountBuilder<'a, Chain: CwEnv> {
+    pub(crate) abstr: &'a Abstract<Chain>,
+    name: Option<String>,
+    description: Option<String>,
+    link: Option<String>,
+    namespace: Option<String>,
+    base_asset: Option<AssetEntry>,
+    // TODO: Decide if we want to abstract this as well.
+    governance_details: Option<GovernanceDetails<String>>,
+    // TODO: How to handle install_modules?
+}
+
+impl<'a, Chain: CwEnv> AccountBuilder<'a, Chain> {
+    pub(crate) fn new(abstr: &'a Abstract<Chain>) -> Self {
+        Self {
+            abstr,
+            name: None,
+            description: None,
+            link: None,
+            namespace: None,
+            base_asset: None,
+            governance_details: None,
+        }
+    }
+
+    pub fn name(self, name: impl Into<String>) -> Self {
```

Comment:
In general it's considered best-practice to use `&mut self` for builders: 
https://blog.logrocket.com/build-rust-api-builder-pattern/

Reply:
Sure will change, I wasn't really sure which was best in this case.